web-api/src/ingestion/preprocess_document.rs

```diff
@@ -53,3 +55,21 @@ fn embed_with_summarizer(
     let embedding = embedder.run(&summary)?;
     Ok(vec![(snippet.clone(), embedding)])
 }
+
+fn embed_with_cutters(
+    embedder: &Embedder,
+    snippet: &DocumentSnippet,
+) -> Result<DocumentContent, Error> {
+    let mut snippets = Vec::new();
+    for split in cutters::cut(snippet, cutters::Language::English) {
+        let str = split.str.trim();
+        let snippet = DocumentSnippet::new(str, str.len())?;
+        let embedding = embedder.run(split.str)?;
```

Comment:
🔧 trimming is already done by the snippet constructor. and we should compute the embedding on the `snippet` instead of the `str`, otherwise this could introduce bugs if the snippet constructor gets changed.

❓/:wrench: i'm not sure why the `str.len()` is passed to the snippet constructor? since we enforce the max byte size on the input snippet and the splitter must yield shorter snippets, we can pass the original byte size requirement.

```suggestion
        let snippet = DocumentSnippet::new(split.str, snippet.len())?;
        let embedding = embedder.run(&snippet)?;
```

Reply:
I pass len because in this part of the pipeline it's not supposed to fail, neither is the config of the input length limits  supposed to be wired through to here.

But we have to rethink this whole snippet length validation (and payment) anyway (in a different PR).

1. when we accept document which are split we probably want to allow them to be much larger then any specific snippet
2. but at that point  it's _our_ responsibility to keep snippets reasonably small
3.  and if we can't do so for a "small" subset of snippets of the same document we still probably don't want to fail ingesting the whole document I would guess, that would be horrible ux ("hy we reject this document due to arbitrary internal reasons"), no idea what we would do but probably something on the line of if we can't get it smaller summarize it and if that doesn't work truncate it and warn. Anyway outside of this PR.
4. through either-way "small" here is now more dependent on what the tokenizer can handle then storage constraints we have now